### PR TITLE
refactor(daemon): use context to set envs

### PR DESCRIPTION
**Pull request by bwanglzu**
use `context` to pass `envs` to YAML configuration, tests adjusted to new style envs setting in YAML.

### CODE REVIEW

**Comment by bwanglzu** [comment]
seems i break a lot of other tests

File: tests/daemon/unit/api/flow_with_env.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+jtype: Flow
+with:
+  name: my-flow
+  grpc_data_requests: true
+  port_expose: 12345
+  protocol: http
+  env:
+    key1: ${{ context_var1 }}  # this should be replaced
+    key2: $context_var2  # this should not be replaced
+    key3: ${ context_var3 }  # this should not be replaced
```
Comment (deepankarm) [inline tests/daemon/unit/api/flow_with_env.yml:10]: User's Flow yaml might include the first 2 syntaxes. Is there any reason, the 2nd one shouldn't be replaced?

Comment (bwanglzu) [reply tests/daemon/unit/api/flow_with_env.yml:10]: it is not supported by `load_config` while testing. maybe I can allow 2nd case happen

Comment (deepankarm) [reply tests/daemon/unit/api/flow_with_env.yml:10]: But the 2nd syntax is available via an environment variable?

Comment (bwanglzu) [reply tests/daemon/unit/api/flow_with_env.yml:10]: i created a minimum example:

```python
from jina.jaml import JAML
from pprint import pprint

with open('toy.yml')as fp:
    a = JAML.load(fp)
    b = JAML.expand_dict(a, context={'context_var1': 'MMM', 'context_var2': 'WWW'})
pprint(b)
```

with YAML:
```yaml
jtype: Flow
with:
  name: my-flow
  grpc_data_requests: true
  port_expose: 12345
  protocol: http
  random_key: ${{ context_var1 }}
  random_key2: $context_var2
executors:
  - name: local_replicas
    port_in: 45678
    replicas: 2

```

Comment (bwanglzu) [reply tests/daemon/unit/api/flow_with_env.yml:10]: essentially, flow.load_config is calling this `expand_dict` where we pass a `context`

**Review by deepankarm — APPROVED** [review]

**Pull request merged**